frontend_csv/src/parser/parser.rs

```diff
@@ -0,0 +1,230 @@
+                    ));
+            }
+            CSVTokenType::LiteralString => {
+                self.lnmsg_buffer
+                    .as_mut()
+                    .unwrap()
+                    .msg_data
+                    .push(LNMsData::TLVinit(
+                        msg_data_type.val.to_string(),
+                        msg_data_name,
+                    ))
+            }
+            _ => panic!("Unknown Token {:?}", msg_data_type),
+        }
+    }
+
+    /// PArse a TLV type declaration
+    fn parse_tlv_typ(&mut self, tokens: &Vec<CSVToken>) {
+        // init_tlvs,networks,1
+        match self.peek(&tokens).ty {
+            CSVTokenType::LiteralString => {
+                let tls_name = self.pop(&tokens).val.to_string();
+                let tlv_name = self.pop(tokens).val.to_string();
+                let tlv_type = self.pop(&tokens).val.parse::<u64>().unwrap();
+                self.lntlv_buffer = Some(LNTlvType {
+                    tls_type: tls_name,
+                    tlv_name: tlv_name,
+                    tlv_type: tlv_type,
+                    tlv_data: None,
+                });
+            }
+            _ => panic!("Unknown Token {:?}", self.peek(&tokens)),
+        }
+    }
+
+    // tlvdata,init_tlvs,networks,chains,chain_hash,...
+    /// parse TLV data entry
+    fn parse_tlv_data(&mut self, tokens: &Vec<CSVToken>) {
+        self.pop(&tokens); // pop tls name
+        self.pop(&tokens); // pop tlv name
+
+        self.lntlv_buffer.as_mut().unwrap().tlv_data = Some(LNTlvData {
+            name: self.pop(&tokens).val.to_string(),
+            value: self.pop(&tokens).val.to_string(),
+        });
+        if self.peek(&tokens).ty == CSVTokenType::Dotdotdot {
+            self.pop(tokens);
+        }
+    }
+
+    fn insert_and_reset_lnmsg(&mut self) {
+        if self.lnmsg_buffer != None {
+            self.simple_table.insert(
+                self.lnmsg_buffer.clone().unwrap().msg_name.clone(),
+                self.lnmsg_buffer.clone().unwrap(),
+            );
+            self.lnmsg_buffer = None
+        }
+    }
+
+    fn insert_and_reset_tlv(&mut self) {
+        if self.lntlv_buffer != None {
+            self.lnmsg_buffer
+                .as_mut()
+                .unwrap()
+                .tlv_stream
+                .push(self.lntlv_buffer.clone().unwrap());
+            self.lntlv_buffer = None
+        }
+    }
+
+    /// Entry point of the parser!
+    pub fn parse(&mut self, tokens: &Vec<CSVToken>) {
+        while self.peek(&tokens).ty != CSVTokenType::EOF {
+            match self.pop(&tokens).ty {
+                CSVTokenType::MsgTy => {
+                    self.insert_and_reset_lnmsg();
+                    self.parse_msg_typ(&tokens)
+                }
+                CSVTokenType::MsgData => self.parse_msg_data(&tokens),
+                CSVTokenType::TlvType => {
+                    self.insert_and_reset_tlv();
+                    self.parse_tlv_typ(&tokens)
```

Comment:
this sound wired to me but I do not have in mind the code and the csv format, I will go again over this

Reply:
Ok! I'm trying to remove this insert and reset tlv and I'm trying to read it all in one shot https://github.com/lnspec-tools/lncodegen.rs/tree/macros/fix_parser but I need to debug it a little bit